cns/client/client_test.go

```diff
@@ -2145,3 +2159,98 @@ func TestNMASupportedAPIs(t *testing.T) {
 		})
 	}
 }
+
+func TestGetAllNCsFromCns(t *testing.T) {
+	emptyRoutes, _ := buildRoutes(defaultBaseURL, clientPaths)
+	exp := cns.GetAllNetworkContainersResponse{
+		NetworkContainers: nil,
+		Response: cns.Response{
+			ReturnCode: types.Success,
+			Message:    "",
+		},
+	}
+	mockDo := &mockdo{
+		errToReturn:            nil,
+		objToReturn:            cns.GetAllNetworkContainersResponse{},
+		httpStatusCodeToReturn: http.StatusOK,
+	}
+	client := &Client{
+		client: mockDo,
+		routes: emptyRoutes,
+	}
+
+	got, err := client.GetAllNCsFromCns(context.TODO())
+	assert.NoError(t, err)
+	assert.Equal(t, exp.Response.ReturnCode, got.Response.ReturnCode)
+}
+
+func TestPostAllNCsToCns(t *testing.T) {
+	// Create the routes necessary for a test client
+	emptyRoutes, _ := buildRoutes(defaultBaseURL, clientPaths)
+
+	createNCTests := []struct {
+		name      string
+		client    *RequestCapture
+		req       cns.PostNetworkContainersRequest
+		expReq    *cns.PostNetworkContainersRequest
+		shouldErr bool
+	}{
+		{
+			"Empty request",
+			&RequestCapture{
+				Next: &mockdo{},
+			},
+			cns.PostNetworkContainersRequest{},
+			nil,
+			true,
+		},
+		{
+			"Happy path",
+			&RequestCapture{
+				Next: &mockdo{},
+			},
+			postAllNcsRequests,
+			&postAllNcsRequests,
+			false,
+		},
+	}
+
+	for _, test := range createNCTests {
+		test := test
+		t.Run(test.name, func(t *testing.T) {
+			t.Parallel()
```

Comment:
This is where there would be a hazard using an uncontrolled package variable for your test cases. Because the package variable is so easy and convenient to reuse, someone *will* eventually reuse it. If they write to it in a Parallel subtest, then we have a data race in the tests.

Keep the parallel subtests, because I don't think there's any reason why they _can't_ be run in parallel (and fast tests are good). But we need to make sure that the data is sound.

Reply:
Removed `t.Parallel()` and the tests works.

Reply:
@bohuini that's fine too... as I said, I do *like* using `t.Parallel()` wherever possible. Unit tests should be blazing fast (less than 10s for the entire suite IMO), and `t.Parallel()` helps a lot with this. I don't think it'll matter a ton here, so don't feel obligated to add it back. I just wanted to clarify my position.